sdk/identity/azure-identity/test/ut/default_azure_credential_test.cpp

```diff
@@ -196,48 +196,51 @@ TEST(DefaultAzureCredential, LogMessages)
             "authorityHost gets created.");
 
         EXPECT_EQ(log[3].first, Logger::Level::Informational);
-        EXPECT_EQ(log[3].second, "Identity: WorkloadIdentityCredential was created successfully.");
+        EXPECT_EQ(log[3].second, "Identity: ClientAssertionCredential was created successfully.");
```

Comment:
When user creates `WorkloadIdentityCredential`, this message should not be in the log, because the fact that `WorkloadIdentityCredential` is implemented via `ClientAssertionCredential` is its internal detail. Instead, we should use the approach that `DefaultAzureCredential` does to `ChainedTokenCredential` - there is a `ChainedTokenCredentialImpl` constructor which takes `credentialName` parameter, and that credentialName is used in all of its log messages. With `WorkloadIdentityCredential` it should be the same - all interactions with it should produce log messages on behalf of `WorkloadIdentityCredential`, and not of `ClientAssertionCredential`.

Reply:
Addressed. Thanks.